pkg/network/ebpf/c/protocols/protocol-classification-helpers.h

```diff
@@ -230,42 +169,30 @@ static __always_inline void protocol_classifier_entrypoint(struct __sk_buff *skb
         return;
     }
 
-    // Making sure we've not processed the same tcp segment, which can happen when a single packet travels different
-    // interfaces.
-    if (has_sequence_seen_before(&skb_tup, &skb_info)) {
-        return;
-    }
-
-    conn_tuple_t *cached_sock_conn_tup_ptr = bpf_map_lookup_elem(&skb_conn_tuple_to_socket_conn_tuple, &skb_tup);
-    if (cached_sock_conn_tup_ptr == NULL) {
-        return;
-    }
-
-    conn_tuple_t cached_sock_conn_tup = *cached_sock_conn_tup_ptr;
-    conn_tuple_t inverse_skb_conn_tup = {0};
-    invert_conn_tuple(&skb_tup, &inverse_skb_conn_tup);
-    inverse_skb_conn_tup.pid = 0;
-    inverse_skb_conn_tup.netns = 0;
-
-    protocol_t sock_tup_protocol = get_cached_protocol_or_default(&cached_sock_conn_tup);
-    protocol_t inverse_skb_tup_protocol = get_cached_protocol_or_default(&inverse_skb_conn_tup);
-    protocol_t cur_fragment_protocol = choose_protocol(sock_tup_protocol, inverse_skb_tup_protocol);
-
-    // If we've already identified the protocol of the socket, no need to read the buffer and try to classify it.
-    if (cur_fragment_protocol == PROTOCOL_UNCLASSIFIED || cur_fragment_protocol == PROTOCOL_UNKNOWN) {
-        char request_fragment[CLASSIFICATION_MAX_BUFFER];
-        bpf_memset(request_fragment, 0, sizeof(request_fragment));
-        read_into_buffer_for_classification((char *)request_fragment, skb, &skb_info);
-        classify_protocol(&cur_fragment_protocol, request_fragment, sizeof(request_fragment));
-    }
-
-    log_debug("[protocol_classifier_entrypoint]: Classifying protocol as: %d\n", cur_fragment_protocol);
-    // If there has been a change in the classification, save the new protocol.
-    if (sock_tup_protocol != cur_fragment_protocol) {
-        bpf_map_update_with_telemetry(connection_protocol, &cached_sock_conn_tup, &cur_fragment_protocol, BPF_ANY);
-    }
-    if (inverse_skb_tup_protocol != cur_fragment_protocol) {
-        bpf_map_update_with_telemetry(connection_protocol, &inverse_skb_conn_tup, &cur_fragment_protocol, BPF_ANY);
+    protocol_t *cur_fragment_protocol_ptr = bpf_map_lookup_elem(&connection_protocol, &skb_tup);
+    if (cur_fragment_protocol_ptr == NULL) {
+        conn_tuple_t inverse_skb_conn_tup = skb_tup;
+        flip_tuple(&inverse_skb_conn_tup);
+
+        cur_fragment_protocol_ptr = bpf_map_lookup_elem(&connection_protocol, &inverse_skb_conn_tup);
```

Comment:
if we're always double indexing, why do we need two lookups?

Reply:
good catch, removed